grpc-bridge/src/main/java/org/jboss/resteasy/grpc/util/DateUtils.java

```diff
@@ -0,0 +1,94 @@
+package org.jboss.resteasy.grpc.util;
+
+import java.text.ParsePosition;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.TimeZone;
+
+/**
+ * Utility for parsing and generating dates
+ *
+ * @author Stuart Douglas (from undertow)
+ */
+public class DateUtils {
+
+   private static final Locale LOCALE_US = Locale.US;
+
+   private static final TimeZone GMT_ZONE = TimeZone.getTimeZone("GMT");
+
+   private static final String RFC1123_PATTERN = "EEE, dd MMM yyyy HH:mm:ss z";
+
+   /**
+    * Thread local cache of this date format. This is technically a small memory leak, however
+    * in practice it is fine, as it will only be used by server threads.
+    * <p>
+    * This is the most common date format, which is why we cache it.
+    */
+   private static final ThreadLocal<SimpleDateFormat> RFC1123_PATTERN_FORMAT = new ThreadLocal<SimpleDateFormat>() {
+      @Override
+      protected SimpleDateFormat initialValue() {
+         SimpleDateFormat df = new SimpleDateFormat(RFC1123_PATTERN, LOCALE_US);
+         return df;
+      }
+   };
```

Comment:
We should really be using the new date utilities (JSR-310). This would likely not require this thread local.

Reply:
ServletConfigWrapper is gone